### DISCUSSION TITLE
Modify HTML report - add screenshot

### DISCUSSION THREAD

**Discussion by alexyz41** [GoogleChrome/lighthouse#12380] [General] [answered]
Hello!!

First of all I would like to thank all the people that is part of this project. You guys are doing an awesome job.

I am rather new to lighthouse, so I apologise for my ignorance.

I am using lighthouse from the Windows PowerShell with the command

`lighthouse https://mywebsite.com/ --locale=es`

I need to modify the HTML report that the tool generates and attach a screenshot of the webpage and the URL of the website at the top of the report.

![lightshot](https://user-images.githubusercontent.com/63423163/115394463-16607380-a1e3-11eb-8048-f628f87d9bc0.jpg)

Is that possible?

Thanks

**Comment by adamraine** [comment, marked as answer]
It's definitely possible, you just need to modify the HTML report manually to add the picture you want. There are some screenshots used in the timeline view under performance if that helps.

**Reply by alexyz41** [reply]
Thanks for replying adamraine!

your suggestion is interesting, but I am going to bulk generate 100+ reports for my clients, so the manual route is not feasible. Ideally I need to modify the HTML template to get all reports with the desired modifications.

I am not sure in what folder the HTML template is. Any help would be appreciated.

**Reply by adamraine** [reply]
Ah ok. Check out the report generator then https://github.com/GoogleChrome/lighthouse/tree/master/lighthouse-core/report. You should be able to modify the code in that folder to suit your needs.

**Reply by alexyz41** [reply]
That is awesome!! Now I have a clear idea on where I should look for the files to modify.

My doubt now is what file should I actually modify in order to accomplish my goal, and what should I add/remove. 

I know that maybe that's much to ask for, so I would really appreciate any hint or code draft.

Thanks

**Reply by adamraine** [reply]
Here is something quick and easy that you can start with:
```html
<div style="height: 200px; background-image: var(--element-screenshot-url);"></div>
```
Put that element at the top of the `div.lh-container` that has `--element-screenshot-url` defined in the style attribute. Result looks like this:

<img width="422" alt="Screen Shot 2021-04-21 at 3 21 58 PM" src="https://user-images.githubusercontent.com/6752989/115609261-52fc9f80-a2b5-11eb-8f5e-dc9f9d069841.png">

**Reply by alexyz41** [reply]
I really appreciate your help adamraine. I am very close to what I want to acomplish.

I am facing a problem tough. I know that the file that I want to modify is "templates.html" inside \lighthouse\lighthouse-core\report\html.

In the HTML report generated, As you said, the section I want to insert the screenshot is before "lh-container"

![Screenshot_1](https://user-images.githubusercontent.com/63423163/115628793-72b9b500-a301-11eb-8875-109f07dd692a.jpg)

But then, I go to the file "templates.html" and I am not able to find that section

![Screenshot_2](https://user-images.githubusercontent.com/63423163/115628821-8238fe00-a301-11eb-8723-92957c132f55.jpg)

Can you tell me where can I find the "lh-container" section?

Thank you so much.

**Reply by adamraine** [reply]
Not everything is in the template, some is generated by JS.

Look in https://github.com/GoogleChrome/lighthouse/tree/master/lighthouse-core/report/html/renderer/report-renderer.js for how the element gets added.

**Comment by ritesh-chauhan0x1** [comment]
Yes ✅ it’s possible to customize the Lighthouse HTML report and add a screenshot + the website URL at the top.
But Lighthouse does not have a built-in flag for that — you’ll need to post-process the generated report or use Lighthouse programmatically.

Option 1: Post-process the HTML file (quick hack)
Run Lighthouse normally:
lighthouse https://mywebsite.com/ --locale=es --output html --output-path=report.html

Open report.html in a text editor (VS Code).
Right after the <body> tag, insert something like:
<div style="text-align:center; margin-bottom:20px;">
  <img src="screenshot.png" alt="Website Screenshot" style="max-width:100%; border:1px solid #ccc;" />
  <p><strong>URL:</strong> https://mywebsite.com/</p>
</div>

Save → open report.html in browser.
This way, the screenshot and URL will appear above the report.

Save → open report.html in browser.
This way, the screenshot and URL will appear above the report.

Option 3: Use --output json and build a custom report
Run:
lighthouse https://mywebsite.com/ --locale=es --output json --output-path=report.json
Use Lighthouse Viewer or custom HTML generator to render JSON → HTML, and add your screenshot wherever you like.

👉 If you just want a one-time simple solution, Option 1 (manual HTML edit) is fastest.
👉 If you want to repeat this every time, go with Option 2 (Node + Puppeteer automation).
